src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Tables/WorkitemQueryTag.sql

```diff
@@ -0,0 +1,24 @@
+﻿/*************************************************************
+    Workitem Query Tag Table
+    Stores static workitem indexed tags
+    TagPath is represented with delimiters to repesent multiple levels
+    QueryStatus can be 0, or 1 to represent Disabled or Enabled.
+**************************************************************/
+CREATE TABLE dbo.WorkitemQueryTag (
+    TagKey                  INT                  NOT NULL, --PK
+    TagPath                 VARCHAR(64)          NOT NULL,
+    TagVR                   VARCHAR(2)           NOT NULL,
+    QueryStatus             TINYINT              DEFAULT 1 NOT NULL,
```

Comment:
How will we use this? We should add it only when we are ready to use it.

Reply:
QueryStatus is used for TransactionUid, which cannot be queried or returned. The alternative is to put it in the Workitem table, but we wanted to have minimal data there.

Reply:
Thanks Josiah, can you help me understand more on how QueryStatus is use with TransactionUID. Maybe a doc link?

Reply:
There is no direct that explains Transaction UID. This link has all the transaction. PS3.18 (nema.org)



For Change Workitem State



Transaction UID (0008,1195)
The request payload shall include a Transaction UID. The user agent creates the Transaction UID when requesting a transition to the IN PROGRESS state for a given Workitem. The user agent provides that Transaction UID in subsequent transactions with that Workitem.



Retrieve Workitem Transaction



- The returned Workitem shall not contain the Transaction UID (0008,1195) Attribute. This is necessary to preserve this Attribute's role as an access lock.



- If the Workitem is in the IN PROGRESS state, the returned Workitem shall not contain the Transaction UID (0008,1195) Attribute of the Workitem, since that should only be known to the Owner.



- The origin server shall also supply the Transaction UID Query Parameter, which specifies the Transaction UID of the Workitem to be updated.

Reply:
After some discussion, we decided to store the transactionUID in the Workitem table since it is not querable. We are also removing QueryStatus from WorkitemQueryTag table since it is not necessary now